### PR TITLE
[SponsorBlock] Add support for chapter category

### PR DESCRIPTION
**Pull request by ajayyy**
### Description of your *pull request* and other information

Renamed `_categories` to `_titles` and added in the chapter description when applicable. In tests, renamed the function to be clearer with sponsor chapters also existing separately, but if that's too confusing, I can change it to something else.

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### CODE REVIEW

**Comment by ajayyy** [comment]
okay, need to figure out why tests aren't running locally then...

**Comment by pukkandan** [comment]
I thought the `chapter` category was a paid feature 🤔

Could you give an example URL that has chapters? so I can verify everything is working correctly

**Comment by ajayyy** [comment]
Only in the extension

https://www.youtube.com/watch?v=qD6bPNZRRbQ

File: yt_dlp/postprocessor/modify_chapters.py [hunk 1]
```diff
@@ -295,8 +295,8 @@ def _remove_tiny_rename_sponsors(self, chapters):
                 c.update({
                     'category': category,
                     'categories': cats,
```
Comment (pukkandan) [inline yt_dlp/postprocessor/modify_chapters.py:297]: These should give the short names for sponsor sections

File: yt_dlp/postprocessor/sponsorblock.py [hunk 2]
```diff
@@ -68,12 +72,13 @@ def duration_filter(s):
...
             (start, end), cat = s['segment'], s['category']
+            title = s['description'] if cat == 'chapter' else self.CATEGORIES[cat]
             return {
                 'start_time': start,
                 'end_time': end,
                 'category': cat,
-                'title': self.CATEGORIES[cat],
-                '_categories': [(cat, start, end)]
+                'title': title,
+                '_categories': [(title, start, end)]
```
Comment (pukkandan) [inline yt_dlp/postprocessor/sponsorblock.py:81]: It doesn't really matter what this field is called, So I've reverted it to keep the diff small

File: yt_dlp/postprocessor/sponsorblock.py [hunk 3]
```diff
@@ -68,12 +72,13 @@ def duration_filter(s):
 
         def to_chapter(s):
             (start, end), cat = s['segment'], s['category']
+            title = s['description'] if cat == 'chapter' else self.CATEGORIES[cat]
```
Comment (pukkandan) [inline yt_dlp/postprocessor/sponsorblock.py:75]: To clarify, `description` is present if and only if categories is 'chapter', correct? Is there any case where a chapter doesn't have description or where a sponsor has one?

Comment (pukkandan) [reply yt_dlp/postprocessor/sponsorblock.py:75]: The API says "title for chapters, empty string for other segments", so I'll go by that

**Pull request merged**